File: algorithm/leetcode/88_Merge_Sorted_Array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int position = m + n - 1;
        int first = m - 1;
        int second = n - 1;
        while(position >= 0) {
            if(first < 0) {
                nums1[position--] = nums2[second--];
            }else if(second < 0) {
                nums1[position--] = nums1[first--];
            }else {
                nums1[position--] = nums1[first] > nums2[second] ? nums1[first--] : nums2[second--];
            }
        }
    }
}

File: algorithm/leetcode/88_Merge_Sorted_Array.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    nums1.length = m;
    nums2.length = n;
    Array.prototype.push.apply(nums1, nums2)
    nums1.sort((a ,b) => {
        return a - b
    })
};

let i = 5 
let arr = [1,2,3,4,5,6]

console.log(arr[--i], i)